ledger/src/ui/test/attestation/test_attestation.c

```diff
@@ -0,0 +1,385 @@
+#define PRIVATE_KEY                                                            \
+    "\xce\x1c\xa0\xf8\xf6\x40\x4b\xfd\x7a\x0a\x33\x80\xd3\x8c\x21\x5c\x89\x31" \
+    "\x61\x91\xb5\x45\x62\xd8\x32\x68\xca\x57\x01\x73\x30\xfc"
+#define SIGNER_HASH                                                            \
+    "\xdd\xa7\x70\x05\x55\xa3\x7b\x75\x34\x29\x1b\x96\x2d\x9f\x41\x41\xb9\x64" \
+    "\x48\xda\xd7\x12\x81\xef\x7c\x2d\x61\x49\x4c\xcb\xb8\x59"
+#define MSG_SIGNATURE                                                          \
+    "\xc0\xc5\xf8\x75\xdc\xdb\x74\x64\x68\x2a\x30\xd5\x44\xc8\x23\xed\xb8\x8e" \
+    "\x73\x31\xc4\xac\x72\x34\xcb\xae\xb0\x56\x07\x13\x6b\x57"
+
+// Global mock variables
+static att_t G_att_ctx;
+static sigaut_signer_t G_signer_info;
+static cx_ecfp_private_key_t G_priv_key;
+static unsigned char G_path[PUBKEY_PATH_LENGTH];
+
+// Global onboarding flag
+const unsigned char N_onboarded_ui[1];
+
+// Helper functions
+void set_public_key(cx_ecfp_public_key_t *pubkey, char *rawkey) {
+    pubkey->W_len = PUBKEYSIZE;
+    memcpy(pubkey->W, rawkey, pubkey->W_len);
+}
+
+void set_private_key(cx_ecfp_private_key_t *privkey, unsigned char *rawkey) {
+    privkey->d_len = KEYLEN;
+    memcpy(privkey->d, rawkey, privkey->d_len);
+}
+
+// cx mocks
+int cx_ecdsa_init_private_key(cx_curve_t curve,
+                              unsigned char *rawkey,
+                              unsigned int key_len,
+                              cx_ecfp_private_key_t *key) {
+    assert(CX_CURVE_256K1 == curve);
+    ASSERT_MEMCMP(rawkey, PRIVATE_KEY, KEYLEN);
+    assert(rawkey == (unsigned char *)G_att_ctx.priv_key_data);
+    assert(key == &G_att_ctx.priv_key);
+    assert(KEYLEN == key_len);
+    set_private_key(key, rawkey);
+    return 0;
+}
+
+int cx_ecfp_generate_pair(cx_curve_t curve,
+                          cx_ecfp_public_key_t *pubkey,
+                          cx_ecfp_private_key_t *privkey,
+                          int keepprivate) {
+    assert(CX_CURVE_256K1 == curve);
+    assert(pubkey == &G_att_ctx.pub_key);
+    assert(privkey == &G_att_ctx.priv_key);
+    ASSERT_MEMCMP(PRIVATE_KEY, privkey->d, KEYLEN);
+    assert(1 == keepprivate);
+    set_public_key(pubkey, PUBLIC_KEY);
+    return 0;
+}
+
+// os mocks
+unsigned int os_endorsement_get_code_hash(unsigned char *buffer) {
+    // Should not be called for the current implementation
+    ASSERT_FAIL();
+}
+
+unsigned int os_endorsement_key2_derive_sign_data(unsigned char *src,
+                                                  unsigned int srcLength,
+                                                  unsigned char *signature) {
+    assert(src == (unsigned char *)G_att_ctx.msg);
+    assert(srcLength == G_att_ctx.msg_offset);
+    assert(signature == APDU_DATA_PTR);
+    memcpy(signature, MSG_SIGNATURE, sizeof(MSG_SIGNATURE) - 1);
+
+    return sizeof(MSG_SIGNATURE) - 1;
+}
+
+void os_perso_derive_node_bip32(cx_curve_t curve,
```

Comment:
This is where the private key data should be set (see my next comment).

Reply:
fixed!